### ISSUE TITLE
triggering zone.onEventDone many times during single zone.run

### ISSUE THREAD

**Issue by PavelPZ** [angular/zone.js#269]
doClick event of this component:

```javascript
import * as core from 'angular2/core';
import {global} from 'angular2/src/facade/lang';

@core.Component({
  selector: 'my-app',
  template: '<h1 (click)="doClick()">Click</h1>'
})
export class AppComponent {

  constructor(private zone: core.NgZone) {//, private http: http.Http) {
    zone.onTurnDone.subscribe(() => console.log('onTurnDone'));
  }

  doClick() {
    global.zone.fork({
      afterTask: function () {
        console.log('After task');
      }
    }).run(() => setTimeout(() => { setTimeout(() => { }, 2000); }, 2000)); //afterTask is called three times
  }
}


```

produced this console log:

```javascript
After task
onTurnDone
After task
onTurnDone
onTurnDone
After task
onTurnDone
onTurnDone
```

first "After task" is printed immediatelly
second "After task" is printed after two seconds
thirds "After task" is printed after another two seconds

------------------
I am using Flux to record x play flux actions (example of such action is "doClick"). For playing recorder actions I need to play them in sync. I try to use zone.js to recognize "action is completed".

Or is there any other possibility to use zone.js or ngZone to recognize "action completed" event?

**Comment by poxrud** [comment]
setTimeout forks a new zone each time. So afterTask will be run every time you have a setTimeout. 
So in your code you get one afterTask for the initial run() and then two more for the two setTimeouts.

**Comment by poxrud** [comment]
@PavelPZ Looks like this is now fixed in 0.6.0

**Comment by PavelPZ** [comment]
Thanks @poxrud. But what is fixed in 0.6.0?

I modify my example slightly, do you think that calling ```onEventDone``` many times during single zone.run is desired behavior?

```javascript
import * as core from 'angular2/core';
import {global} from 'angular2/src/facade/lang';

@core.Component({
  selector: 'my-app',
  template: '<h1 (click)="doClick()">onEventDone</h1>'
})
export class AppComponent {
  constructor(private zone: core.NgZone) {
    zone.onEventDone.subscribe(() => console.log('onEventDone after ' + Math.round((new Date().getTime() - this.tick) / 1000) + ' sec'));
  }
  tick = 0;
  doClick() {
    this.tick = new Date().getTime();
    console.log('doClick start');
    global.zone.run(() => setTimeout(() => { setTimeout(() => { console.log('doClick end'); }, 2000); }, 2000));
  }
}
```
### log:
```javascript
doClick start
onEventDone after 0 sec
onEventDone after 2 sec
onEventDone after 2 sec
doClick end
onEventDone after 4 sec
onEventDone after 4 sec
```
